### PR TITLE
Symfony 4.4.* & 5.0.* (In Review)

### PR DESCRIPTION
**Pull request by elchris**
Hello again,

I don't expect this to get merged, but I still wanted to get the pull request going just for reference,  in case it helps with the work being done on #118, and also to verify supported versions against your CI.

This time around I'm doing it against **master** ;)

I've made a number of improvements over my last pull request, by looking at @rbaarsma 's composer.json file:

- Symfony 4.4.* should **pass all tests** on php 7.2 and php 7.3
- - with: PSR_HTTP_PROVIDER=nyholm
- - with: PSR_HTTP_PROVIDER=zendframework
- Symfony 5.0.* should **pass all tests** on php 7.2 and php 7.3
- - with: PSR_HTTP_PROVIDER=nyholm
- - with: PSR_HTTP_PROVIDER=zendframework

- This branch contains everything from both **master** and **2.x** along with my own fixes to **2.x** to make the tests pass, as submitted in #148 

- update: CI passes fully.

### CODE REVIEW

**Comment by elchris** [comment]
All tests do seem to pass against the desired targets ... If ya'll wanted to roll with this, cool by me.

**Comment by elchris** [comment]
On a separate note, I also have a different [branch](https://github.com/elchris/oauth2-bundle/tree/php-7.4-symfony-5) with your **add-php7.4-testing-target** branch merged with my **symfony-5** branch, but removing all pre-Symfony 4.4 testing targets, and adding Symfony 5 testing targets.

With that branch you could have:

**Symfony 4.4 and 5.0 on PHP 7.2, 7.3, 7.4 with nyholm and zendframework**

So far my local testing seems fine, but I want to keep trying different permutations locally.

Let me know whether you'd eventually be interested in a PR against, I imagine, **master** ?

**Comment by X-Coder264** [comment]
> On a separate note, I also have a different [branch](https://github.com/elchris/oauth2-bundle/tree/php-7.4-symfony-5) with your **add-php7.4-testing-target** branch merged with my **symfony-5** branch, but removing all pre-Symfony 4.4 testing targets, and adding Symfony 5 testing targets.
> 
> With that branch you could have:
> 
> **[Symfony 4.4 and 5.0 on PHP 7.2, 7.3, 7.4 with nyholm and zendframework](https://github.com/elchris/oauth2-bundle/blob/php-7.4-symfony-5/.travis.yml#L10)**

This is the state that is gonna be on `master` as soon as https://github.com/trikoder/oauth2-bundle/pull/134 and https://github.com/trikoder/oauth2-bundle/pull/118 get merged. So I'm not sure what are you actually suggesting as it seems to be the same as the PRs that are already opened?

**Comment by elchris** [comment]
> 
> This is the state that is gonna be on `master` as soon as #134 and #118 get merged. So I'm not sure what are you actually suggesting as it seems to be the same as the PRs that are already opened?

I looked at #118 and it still appears to have a few failing tests, it looks like it's pretty close but it could take some more time, which, I imagine, is perfectly fine.

With this said, for now it would appear that on this branch "symfony-5", I do have all tests passing, unless there's something which I am missing. At first I didn't think I could pull it off, to have it working with both Symfony 4.4 and Symfony 5, which is why I was at first suggesting to just look at this for informational purposes, but in the end it worked out easily. As such it would seem that this branch could be a viable contender for now, and you could in theory consider it as a candidate for merging, although I do appreciate that the work duplication is unfortunate, and it's my fault for not having looked at the other pull requests in the first place, so I'll completely understand if it's better to still close mine out.

After combining your work on PHP 7.4 testing target with my symfony-5 branch, all tests also appear to pass just fine, although I've not yet vetted it against your CI run.

So to answer the question as far was what I'm suggesting, I see a few possibilities:

1. **Stick with the current plan**: keep working on #118 , merge it to master, then fix the tests in 2.x, then merge it to master, then eventually merge the 7.4 testing target branch into master after removing testing targets for Symfony < 4.4
2. **Alternate Plan 1**: Evaluate this pull request for merging into master, which would give you Symfony 4.4 and 5, along with league 8, as well as your 2.x branch with my fixes for the failing tests.
3. **Alternate Plan 2**: I could make a pull request for [this branch](https://github.com/elchris/oauth2-bundle/tree/php-7.4-symfony-5) against master which would, in addition to this pull request, give you the testing targets for PHP 7.4 targeted at Symfony 4.4 and 5. The tests I've been running locally appear to work fine.

**Comment by elchris** [comment]
update: I've referenced my fork on a personal Symfony 5 project I just started, and it seems to be working fine so-far.

composer.json:

`
"repositories" : [
{
     "type": "vcs",
     "url": "https://github.com/elchris/oauth2-bundle"
}
]
`

and then

composer require trikoder/oauth2-bundle dev-symfony-5

File: League/Repository/ClientRepository.php [hunk 1]
```diff
@@ -86,6 +86,6 @@ private function isGrantSupported(ClientModel $client, ?string $grant): bool
             return true;
         }
 
-        return \in_array($grant, $client->getGrants());
+        return \in_array($grant, $client->getGrants(), true);
```
Comment (mrosiu) [inline League/Repository/ClientRepository.php]: `$grant` contains string,
`$client->getGrants()` returns array of `Trikoder\Bundle\OAuth2Bundle\Model\Client`
So we can not use `strict = true` here

Comment (elchris) [reply League/Repository/ClientRepository.php]: oh okay thank you for pointing this out. will fix.

**Comment by elchris** [comment]
I fixed the conflicts.

All CI tests pass against Symfony 4.4 and Symfony 5.0 with PHP 7.2 and PHP 7.3 with both PSR providers.

**Comment by dawidgora** [comment]
So if everything is ok on this PR, maybe it could get merged? Author of #118 said he is ok with merging this instead of his work. It would be cool to have the bundle ready to be used in Symfony 5.0 asap ;)

**Comment by spideyfusion** [comment]
Hang tight everyone, this is our next effort to get up and running on the `master` branch.

@elchris Thank you for your efforts thus far!

**Comment by elchris** [comment]
> Hang tight everyone, this is our next effort to get up and running on the `master` branch.
> 
> @elchris Thank you for your efforts thus far!

awesome, I'll be standing-by if you need anything from me. Thank you for all your work, and good luck :)

**Comment by X-Coder264** [comment]
@elchris Please rebase with master as we just merged v2.x into it.

File: .travis.yml [hunk 2]
```diff
@@ -9,24 +9,17 @@ services:
...
 env:
     # PHP 7.2
-    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=nyholm SYMFONY_VERSION=3.4.* XDEBUG_VERSION=2.7.2
-    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=nyholm SYMFONY_VERSION=4.2.* XDEBUG_VERSION=2.7.2
-    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=nyholm SYMFONY_VERSION=4.3.* XDEBUG_VERSION=2.7.2
     - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=nyholm SYMFONY_VERSION=4.4.* XDEBUG_VERSION=2.7.2
-    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=zendframework SYMFONY_VERSION=3.4.* XDEBUG_VERSION=2.7.2
-    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=zendframework SYMFONY_VERSION=4.2.* XDEBUG_VERSION=2.7.2
-    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=zendframework SYMFONY_VERSION=4.3.* XDEBUG_VERSION=2.7.2
+    - PHP_VERSION=7.2 PSR_HTTP_PROVIDER=nyholm SYMFONY_VERSION=5.0.* XDEBUG_VERSION=2.7.2
```
Comment (X-Coder264) [inline .travis.yml]: After you rebase with master there shouldn't be any `XDEBUG_VERSION` here left and `SYMFONY_VERSION` became `SYMFONY_REQUIRE`.

File: Manager/Doctrine/AccessTokenManager.php [hunk 3]
```diff
@@ -26,7 +26,10 @@ public function __construct(EntityManagerInterface $entityManager)
      */
     public function find(string $identifier): ?AccessToken
     {
-        return $this->entityManager->find(AccessToken::class, $identifier);
```
Comment (X-Coder264) [inline Manager/Doctrine/AccessTokenManager.php]: This change seems completely unrelated with Symfony 5.

File: Manager/Doctrine/RefreshTokenManager.php [hunk 4]
```diff
@@ -26,7 +26,10 @@ public function __construct(EntityManagerInterface $entityManager)
      */
     public function find(string $identifier): ?RefreshToken
     {
-        return $this->entityManager->find(RefreshToken::class, $identifier);
```
Comment (X-Coder264) [inline Manager/Doctrine/RefreshTokenManager.php]: Same as above.

File: Model/AccessToken.php [hunk 5]
```diff
@@ -40,7 +40,7 @@ class AccessToken
 
     public function __construct(
         string $identifier,
-        DateTimeInterface $expiry,
+        DateTimeImmutable $expiry,
```
Comment (X-Coder264) [inline Model/AccessToken.php]: Why was this changed? It doesn't seem to be related with SF 5.

File: Security/Firewall/OAuth2Listener.php [hunk 6]
```diff
@@ -53,12 +52,12 @@ public function __construct(
     /**
      * BC layer for Symfony < 4.3
      */
-    public function handle(GetResponseEvent $event)
+    public function handle(RequestEvent $event)
```
Comment (X-Coder264) [inline Security/Firewall/OAuth2Listener.php]: We don't need the handle method anymore since that was the BC layer for Symfony < 4.3 which we won't support on master after this gets merged,

File: Tests/Acceptance/AuthorizationEndpointTest.php [hunk 7]
```diff
@@ -5,15 +5,16 @@
...
 use DateTimeImmutable;
+use Nyholm\Psr7\Response;
 use Trikoder\Bundle\OAuth2Bundle\Event\AuthorizationRequestResolveEvent;
 use Trikoder\Bundle\OAuth2Bundle\Manager\AccessTokenManagerInterface;
 use Trikoder\Bundle\OAuth2Bundle\Manager\AuthorizationCodeManagerInterface;
 use Trikoder\Bundle\OAuth2Bundle\Manager\ClientManagerInterface;
 use Trikoder\Bundle\OAuth2Bundle\Manager\RefreshTokenManagerInterface;
 use Trikoder\Bundle\OAuth2Bundle\Manager\ScopeManagerInterface;
 use Trikoder\Bundle\OAuth2Bundle\OAuth2Events;
+//use Zend\Diactoros\Response;
```
Comment (X-Coder264) [inline Tests/Acceptance/AuthorizationEndpointTest.php]: ?

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 8]
```diff
@@ -72,7 +74,7 @@ private function buildAccessToken(string $identifier, string $modify, Client $cl
     {
         return new AccessToken(
             $identifier,
-            new DateTimeImmutable($modify),
+            (new DateTimeImmutable())->modify($modify),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 9]
```diff
@@ -138,10 +140,10 @@ private function buildRefreshToken(string $identifier, string $modify, Client $c
     {
         return new RefreshToken(
             $identifier,
-            new DateTimeImmutable('+1 day'),
+            (new DateTimeImmutable('+1 day')),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/DoctrineClientManagerTest.php [hunk 10]
```diff
@@ -46,7 +46,7 @@ public function testClientDeleteCascadesToAccessTokens(): void
         $em->persist($client);
         $em->flush();
 
-        $accessToken = new AccessToken('access token', new DateTimeImmutable('+1 day'), $client, $client->getIdentifier(), []);
+        $accessToken = new AccessToken('access token', (new DateTimeImmutable())->modify('+1 day'), $client, $client->getIdentifier(), []);
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineClientManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/DoctrineClientManagerTest.php [hunk 11]
```diff
@@ -79,11 +79,11 @@ public function testClientDeleteCascadesToAccessTokensAndRefreshTokens(): void
...
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineClientManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/DoctrineClientManagerTest.php [hunk 12]
```diff
@@ -79,11 +79,11 @@ public function testClientDeleteCascadesToAccessTokensAndRefreshTokens(): void
...
         $em->persist($accessToken);
         $em->flush();
 
-        $refreshToken = new RefreshToken('refresh token', new DateTimeImmutable('+1 day'), $accessToken);
+        $refreshToken = new RefreshToken('refresh token', (new DateTimeImmutable())->modify('+1 day'), $accessToken);
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineClientManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/DoctrineRefreshTokenManagerTest.php [hunk 13]
```diff
@@ -75,10 +77,10 @@ private function buildRefreshToken(string $identifier, string $modify, Client $c
     {
         return new RefreshToken(
             $identifier,
-            new DateTimeImmutable($modify),
+            (new DateTimeImmutable())->modify($modify),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineRefreshTokenManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/DoctrineRefreshTokenManagerTest.php [hunk 14]
```diff
@@ -75,10 +77,10 @@ private function buildRefreshToken(string $identifier, string $modify, Client $c
...
             new AccessToken(
                 $identifier,
-                new DateTimeImmutable('+1 day'),
+                (new DateTimeImmutable('+1 day')),
```
Comment (X-Coder264) [inline Tests/Acceptance/DoctrineRefreshTokenManagerTest.php]: This change seems unrelated to SF 5.

File: Tests/Acceptance/TokenEndpointTest.php [hunk 15]
```diff
@@ -20,12 +20,22 @@ protected function setUp(): void
     {
         parent::setUp();
 
+        /** @var ScopeManagerInterface $scopeManager */
+        $scopeManager = $this->client->getContainer()->get(ScopeManagerInterface::class);
```
Comment (X-Coder264) [inline Tests/Acceptance/TokenEndpointTest.php]: All changes here are unrelated to Symfony 5.

File: Tests/Fixtures/SecurityTestController.php [hunk 16]
```diff
@@ -28,11 +28,9 @@ public function scopeAction(): Response
 
     public function rolesAction(): Response
     {
-        $roles = $this->get('security.token_storage')->getToken()->getRoles();
-
-        $roles = array_map(function (Role $role): string {
-            return $role->getRole();
-        }, $roles);
+        /** @var TokenStorage $tokenStorage */
+        $tokenStorage = $this->get('security.token_storage');
```
Comment (X-Coder264) [inline Tests/Fixtures/SecurityTestController.php]: This change seems unrelated to SF 5.

File: Tests/TestHelper.php [hunk 17]
```diff
@@ -44,9 +46,17 @@ public static function generateEncryptedPayload(RefreshTokenModel $refreshToken)
...
+        if (\count($uris) > 0) {
+            $uri = $uris[0];
+        } else {
+            $uri = new RedirectUri('http://localhost/');
+        }
+
         $payload = json_encode([
             'client_id' => $authCode->getClient()->getIdentifier(),
-            'redirect_uri' => (string) $authCode->getClient()->getRedirectUris()[0],
+            'redirect_uri' => (string) $uri,
```
Comment (X-Coder264) [inline Tests/TestHelper.php]: This change seems unrelated to SF 5.

File: Tests/TestKernel.php [hunk 18]
```diff
@@ -56,11 +65,11 @@ public function boot()
     public function registerBundles()
     {
         return [
-            new \Doctrine\Bundle\DoctrineBundle\DoctrineBundle(),
-            new \Sensio\Bundle\FrameworkExtraBundle\SensioFrameworkExtraBundle(),
-            new \Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
-            new \Symfony\Bundle\SecurityBundle\SecurityBundle(),
-            new \Trikoder\Bundle\OAuth2Bundle\TrikoderOAuth2Bundle(),
+            new DoctrineBundle(),
```
Comment (X-Coder264) [inline Tests/TestKernel.php]: These changes are unrelated to SF 5.

File: Tests/Unit/ExtensionTest.php [hunk 19]
```diff
@@ -79,14 +78,21 @@ private function getValidConfiguration(array $options): array
                 'resource_server' => [
                     'public_key' => 'foo',
                 ],
-                'persistence' => [],
+                //Pick one for valid config:
+                //'persistence' => ['doctrine' => []]
+                'persistence' => ['in_memory' => 1],
```
Comment (X-Coder264) [inline Tests/Unit/ExtensionTest.php]: This change seems unrelated to Symfony 5.

File: Tests/Unit/ExtensionTest.php [hunk 20]
```diff
@@ -79,14 +78,21 @@ private function getValidConfiguration(array $options): array
...
     private function setupContainer(ContainerBuilder $container): void
     {
         $container->register(ScopeManager::class);
-        $container->setAlias(ScopeManagerInterface::class, ScopeManager::class);
+        //the line below is no-longer needed if "persistence" above
+        //is set to either doctrine or in_memory
+        //which will ensure that services defined in either
+        // storage/in_memory.xml or storage/doctrine.xml
+        //get properly defined
+        //$container->setAlias(ScopeManagerInterface::class, ScopeManager::class);
```
Comment (X-Coder264) [inline Tests/Unit/ExtensionTest.php]: This change seems unrelated to Symfony 5.

File: Tests/Unit/InMemoryAccessTokenManagerTest.php [hunk 21]
```diff
@@ -62,7 +62,7 @@ private function buildAccessToken(string $identifier, string $modify): AccessTok
     {
         return new AccessToken(
             $identifier,
-            new DateTimeImmutable($modify),
+            (new DateTimeImmutable())->modify($modify),
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryAccessTokenManagerTest.php]: This change is unrelated to Symfony 5.

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php [hunk 22]
```diff
@@ -63,10 +63,10 @@ private function buildRefreshToken(string $identifier, string $modify): RefreshT
     {
         return new RefreshToken(
             $identifier,
-            new DateTimeImmutable($modify),
+            (new DateTimeImmutable())->modify($modify),
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryRefreshTokenManagerTest.php]: This change is unrelated to Symfony 5.

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php [hunk 23]
```diff
@@ -63,10 +63,10 @@ private function buildRefreshToken(string $identifier, string $modify): RefreshT
...
             new AccessToken(
                 $identifier,
-                new DateTimeImmutable('+1 day'),
+                (new DateTimeImmutable('+1 day')),
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryRefreshTokenManagerTest.php]: This change is unrelated to Symfony 5.

File: composer.json [hunk 24]
```diff
@@ -14,23 +14,24 @@
     "require": {
         "php": ">=7.2",
         "doctrine/doctrine-bundle": "^1.8|^2.0",
-        "doctrine/orm": "^2.6",
+        "doctrine/orm": "^2.7|^3.0",
```
Comment (X-Coder264) [inline composer.json]: The constraint should be just `^2.7`. We can't claim to support 3.0 as it hasn't even been released yet and who knows what breaking changes will yet be made there.

File: composer.json [hunk 25]
```diff
@@ -14,23 +14,24 @@
...
         "league/oauth2-server": "^8.0",
         "psr/http-factory": "^1.0",
-        "sensio/framework-extra-bundle": "^5.3",
-        "symfony/framework-bundle": "^3.4|^4.2",
-        "symfony/psr-http-message-bridge": "^1.2",
-        "symfony/security-bundle": "^3.4|^4.2"
+        "sensio/framework-extra-bundle": "^5.5",
+        "symfony/psr-http-message-bridge": "^1.3",
+        "symfony/security-bundle": "^4.4|^5.0",
+        "symfony/framework-bundle": "^4.4|^5.0.2"
```
Comment (X-Coder264) [inline composer.json]: What is the reason that 5.0.2 is picked as the minimum version in the 5.x series instead of just 5.0?

File: composer.json [hunk 26]
```diff
@@ -14,23 +14,24 @@
...
-        "nyholm/psr7": "^1.1",
-        "phpunit/phpunit": "^8.4",
-        "symfony/browser-kit": "^3.4|^4.2",
-        "symfony/phpunit-bridge": "^3.4|^4.4",
-        "zendframework/zend-diactoros": "^2.1"
+        "friendsofphp/php-cs-fixer": "^2.16",
+        "laminas/laminas-diactoros": "^2.2",
+        "nyholm/psr7": "^1.2",
+        "phpunit/phpunit": "^8.5",
+        "roave/security-advisories": "dev-master",
```
Comment (X-Coder264) [inline composer.json]: This addition is unrelated to Symfony 5.

File: composer.json [hunk 27]
```diff
@@ -55,7 +56,7 @@
             "dev-master": "3.x-dev"
         },
         "symfony": {
-            "require": "4.4.*"
+            "require": "4.4.*|5.0.*"
```
Comment (X-Coder264) [inline composer.json]: This was removed on master so after you rebase this should be deleted.

**Comment by X-Coder264** [comment]
The `getWrappedTreeBuilder` method of the `\Trikoder\Bundle\OAuth2Bundle\DependencyInjection\Configuration ` can also be removed now since that was there to keep BC with Symfony 3.4.

File: Tests/Fixtures/SecurityTestController.php (hunk 16 above)
Comment (X-Coder264) [inline Tests/Fixtures/SecurityTestController.php]: There is no need to get the token storage in this way. Use constructor or method injection instead,

**Comment by elchris** [comment]
just merged upstream/master, gotta work thru the requested changes now. i may need a day or two :)

File: Manager/Doctrine/AccessTokenManager.php (hunk 3 above)
Comment (elchris) [reply Manager/Doctrine/AccessTokenManager.php]: i'm happy to revert those changes but ... are you sure that you want me to? the arguments were referencing those classes as "string paths" and changing those references to the ":class" syntax promotes better static analysis, which is immensely helpful when undertaking framework upgrades like this one. It definitely helped me in the process of upgrading for compatibility for symfony 4.4/5.0.

File: Tests/Acceptance/AuthorizationEndpointTest.php (hunk 7 above)
Comment (elchris) [reply Tests/Acceptance/AuthorizationEndpointTest.php]: Zend/Diactoros got deprecated and replaced by the laminas project, but I certainly should remove this comment

File: Manager/Doctrine/AccessTokenManager.php (hunk 3 above)
Comment (X-Coder264) [reply Manager/Doctrine/AccessTokenManager.php]: But this change (adding a temporary variable just to add a docblock) doesn't really improve anything (other than maybe making your IDE happy as technically the `find` method returns `object|null`) . The method return typehint is there, the entityManager gets called with `AccessToken::class` and there should also be tests for this as we have ~90% test coverage (if there isn't there should be one). So I don't see how this change helped you with the Symfony 5 upgrade, especially since this change is just a docblock change which concerns a Doctrine, not a Symfony class.

Comment (elchris) [reply Manager/Doctrine/AccessTokenManager.php]: oh okay yes sure, I wasn't looking at the correct change. ok thank you.

File: .travis.yml (hunk 2 above)
Comment (elchris) [reply .travis.yml]: ok i think this is fixed after the merge from master.

File: Manager/Doctrine/AccessTokenManager.php (hunk 3 above)
Comment (elchris) [reply Manager/Doctrine/AccessTokenManager.php]: ok should be fixed now ...

File: Manager/Doctrine/RefreshTokenManager.php (hunk 4 above)
Comment (elchris) [reply Manager/Doctrine/RefreshTokenManager.php]: should be fixed now.

File: Model/AccessToken.php (hunk 5 above)
Comment (elchris) [reply Model/AccessToken.php]: should be reverted.

File: Security/Firewall/OAuth2Listener.php (hunk 6 above)
Comment (elchris) [reply Security/Firewall/OAuth2Listener.php]: removed it.

File: Tests/Acceptance/AuthorizationEndpointTest.php (hunk 7 above)
Comment (elchris) [reply Tests/Acceptance/AuthorizationEndpointTest.php]: removed comment

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php (hunk 8 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php (hunk 9 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineAccessTokenManagerTest.php [hunk 28]
```diff
@@ -138,10 +140,10 @@ private function buildRefreshToken(string $identifier, string $modify, Client $c
...
             new AccessToken(
                 $identifier,
-                new DateTimeImmutable($modify),
+                (new DateTimeImmutable())->modify($modify),
```
[the remark this answers is not in the document]
Comment (elchris) [reply Tests/Acceptance/DoctrineAccessTokenManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineClientManagerTest.php (hunk 10 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineClientManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineClientManagerTest.php (hunk 11 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineClientManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineClientManagerTest.php (hunk 12 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineClientManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineRefreshTokenManagerTest.php (hunk 13 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineRefreshTokenManagerTest.php]: fixed

File: Tests/Acceptance/DoctrineRefreshTokenManagerTest.php (hunk 14 above)
Comment (elchris) [reply Tests/Acceptance/DoctrineRefreshTokenManagerTest.php]: fixed

File: Tests/Acceptance/TokenEndpointTest.php (hunk 15 above)
Comment (elchris) [reply Tests/Acceptance/TokenEndpointTest.php]: fixed

File: Tests/Fixtures/SecurityTestController.php (hunk 16 above)
Comment (elchris) [reply Tests/Fixtures/SecurityTestController.php]: so this change is needed because TokenStorage changed in SF5, where you no-longer call role->getRole. reverting this change will actually cause an Error in the unit tests. So I do think  we need to keep this one around.

File: Tests/TestHelper.php (hunk 17 above)
Comment (elchris) [reply Tests/TestHelper.php]: it's a similar deal with this change as the other one. reverting these changes will cause an error for undefined offset:

1) Trikoder\Bundle\OAuth2Bundle\Tests\Integration\AuthorizationServerTest::testFailedAuthorizationWithCodeForOtherClient
Undefined offset: 0

/Users/chris/projects/project-oauth2-bundle/oauth2-bundle/Tests/TestHelper.php:51

Comment (elchris) [reply Tests/TestHelper.php]: so i would advise keeping this change in.

File: Tests/TestKernel.php (hunk 18 above)
Comment (elchris) [reply Tests/TestKernel.php]: reverted ...

File: Tests/Unit/ExtensionTest.php (hunk 19 above)
Comment (elchris) [reply Tests/Unit/ExtensionTest.php]: this change was helpful to me though, when trying to understand how to configure the package in my own project, it helped me validate what an in-memory persistence should be configured like. This is an actual proper configuration of the extension, because the test had an artificial manual call to "$container->setAlias(ScopeManagerInterface::class, ScopeManager::class);" which was only making the test pass artificially without signaling what a proper configuration should look like.

I would recommend keeping this.

File: Tests/Unit/ExtensionTest.php (hunk 20 above)
Comment (elchris) [reply Tests/Unit/ExtensionTest.php]: I would recommend keeping this. See my comment on the previous issue.

File: Tests/Unit/InMemoryAccessTokenManagerTest.php (hunk 21 above)
Comment (elchris) [reply Tests/Unit/InMemoryAccessTokenManagerTest.php]: reverted

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php (hunk 22 above)
Comment (elchris) [reply Tests/Unit/InMemoryRefreshTokenManagerTest.php]: reverted

File: Tests/Unit/InMemoryRefreshTokenManagerTest.php (hunk 23 above)
Comment (elchris) [reply Tests/Unit/InMemoryRefreshTokenManagerTest.php]: reverted

File: composer.json (hunk 24 above)
Comment (elchris) [reply composer.json]: fixed, thank you!

File: composer.json (hunk 26 above)
Comment (elchris) [reply composer.json]: - nyholm/psr7: minor point upgrade from master 1.1 to 1.2, i think we should keep to stay up to date in dev (should keep)
- phpunit: minor point upgrade from master 8.4 to 8.5, i think we should keep to stay up to date in dev (should keep)
- laminas: replaces the deprecated zendframework/zend-diactoros package, which would keep throwing warnings in test. I upgraded all tests to reference laminas instead of zendframework (must keep)
- i do think that we should also keep roave/security-advisories in dev because it will help us flag packages with security issues during development. But I can remove it if you really wish. (i can remove)

File: composer.json (hunk 25 above)
Comment (elchris) [reply composer.json]: fixed, my bad :) thank you!

File: composer.json (hunk 27 above)
Comment (elchris) [reply composer.json]: yes, it removed it after merge from master, thank you!

**Comment by elchris** [comment]
> The `getWrappedTreeBuilder` method of the `\Trikoder\Bundle\OAuth2Bundle\DependencyInjection\Configuration` class can also be removed now since that was there to keep BC with Symfony 3.4.

I am not sure what to do here because that method appears to be invoked all over the class. If you or someone could suggest a pull request into my branch, or give me a code snippet for that whole class, i'll be happy to integrate the suggestion

File: Tests/Fixtures/SecurityTestController.php (hunk 16 above)
Comment (elchris) [reply Tests/Fixtures/SecurityTestController.php]: i'm not sure how to achieve this, so feel free to give me a code snippet or a pull request into my branch.

File: composer.json (hunk 26 above)
Comment (X-Coder264) [reply composer.json]: I was talking only about `roave/security-advisories` here. I'd like to keep this PR as minimal as possible and just focus on Symfony 5 support. Stuff like `roave/security-advisories`  can be PR-ed and considered separately. The bumping of other dev dependencies is fine.

File: Tests/Fixtures/SecurityTestController.php (hunk 16 above)
Comment (X-Coder264) [reply Tests/Fixtures/SecurityTestController.php]: You can typehint `TokenStorageInterface` in the constructor or as a param for `rolesAction` method so that there is no need to call `$this->get()` here.

File: Tests/TestHelper.php (hunk 17 above)
Comment (X-Coder264) [reply Tests/TestHelper.php]: This has nothing to do with Symfony 5 and our CI is passing without this changes so I'd like to revert this for this PR. If you can submit a new PR with a test where this fails and with this fix please do so.

File: Model/AccessToken.php (hunk 5 above)
Comment (X-Coder264) [reply Model/AccessToken.php]: It still hasn't been reverted. Also please don't resolve discussions when you push something.

File: composer.json [hunk 29]
```diff
@@ -14,22 +14,24 @@
...
     },
     "require-dev": {
         "ext-timecop": "*",
         "ext-xdebug": "*",
-        "nyholm/psr7": "^1.1",
-        "phpunit/phpunit": "^8.4",
-        "symfony/browser-kit": "^3.4|^4.2",
-        "symfony/phpunit-bridge": "^3.4|^4.4",
-        "zendframework/zend-diactoros": "^2.1"
+        "friendsofphp/php-cs-fixer": "^2.16",
```
Comment (X-Coder264) [inline composer.json]: @elchris This was wrongly rebased. We removed CS fixer as a composer dev dependency as it is now a part of the Docker image as a PHAR.

**Comment by X-Coder264** [comment]
> > The `getWrappedTreeBuilder` method of the `\Trikoder\Bundle\OAuth2Bundle\DependencyInjection\Configuration` class can also be removed now since that was there to keep BC with Symfony 3.4.
> 
> I am not sure what to do here because that method appears to be invoked all over the class. If you or someone could suggest a pull request into my branch, or give me a code snippet for that whole class, i'll be happy to integrate the suggestion

The method should be completely removed and then stuff like:

`$treeBuilder = $this->getWrappedTreeBuilder('trikoder_oauth2');`

should become

`$treeBuilder = new TreeBuilder('trikoder_oauth2');`

File: Model/AccessToken.php (hunk 5 above)
Comment (elchris) [reply Model/AccessToken.php]: oh ok sorry my bad.

Comment (elchris) [reply Model/AccessToken.php]: should be fixed now :)

File: Tests/TestHelper.php (hunk 17 above)
Comment (elchris) [reply Tests/TestHelper.php]: ok done ....

File: composer.json (hunk 26 above)
Comment (elchris) [reply composer.json]: ok I removed roave/security-advisories

File: Tests/Fixtures/SecurityTestController.php (hunk 16 above)
Comment (elchris) [reply Tests/Fixtures/SecurityTestController.php]: ah ok cool, yeah that worked, thank you! :)

File: composer.json (hunk 29 above)
Comment (elchris) [reply composer.json]: ah sorry about that, fixed now, thank you!

**Comment by elchris** [comment]
> > > The `getWrappedTreeBuilder` method of the `\Trikoder\Bundle\OAuth2Bundle\DependencyInjection\Configuration` class can also be removed now since that was there to keep BC with Symfony 3.4.
> > 
> > 
> > I am not sure what to do here because that method appears to be invoked all over the class. If you or someone could suggest a pull request into my branch, or give me a code snippet for that whole class, i'll be happy to integrate the suggestion
> 
> The method should be completely removed and then stuff like:
> 
> `$treeBuilder = $this->getWrappedTreeBuilder('trikoder_oauth2');`
> 
> should become
> 
> `$treeBuilder = new TreeBuilder('trikoder_oauth2');`

Oh ok thank you! Looks like it worked, fixed.

File: Tests/Fixtures/SecurityTestController.php (hunk 16 above)
Comment (X-Coder264) [reply Tests/Fixtures/SecurityTestController.php]: 👍

File: Tests/Unit/ExtensionTest.php (hunk 19 above)
Comment (elchris) [reply Tests/Unit/ExtensionTest.php]: I could revert this and the one below, if you still wish me to, and put it in a later PR. Let me know @X-Coder264

Comment (X-Coder264) [reply Tests/Unit/ExtensionTest.php]: @elchris This is fine actually. :+1:

File: Tests/Unit/ExtensionTest.php (hunk 20 above)
Comment (X-Coder264) [reply Tests/Unit/ExtensionTest.php]: @elchris Sure, just please delete all these commented lines.

Comment (elchris) [reply Tests/Unit/ExtensionTest.php]: ok, done, pushed!

Comment (X-Coder264) [reply Tests/Unit/ExtensionTest.php]: :+1:

**Comment by elchris** [comment]
> @elchris Thank you for the effort that you put into this PR.

And thank you for your patience and guidance! :)
